### ISSUE TITLE
[BUG] Ongoing Loss issues in BioNeMo when finetuning.

### ISSUE THREAD

**Issue by skothenhill-nv** [NVIDIA/bionemo-framework#904]
### BioNeMo Framework Version

Ongoing Loss issues in BioNeMo when finetuning.

### Bug Description

esm2/model/finetune
- RegressorLossReduction doesnt respect tensor parallel (or any other form of parallelism), no reductions happen
- no reduction anywhere (DP), and mse_loss doesnt work with TP- needs a vocab parallel 
- ClassifierLossReduction
- again no reductions, vocab parallel cross entropy for TP, and actual reductions for DDP

- Geneformer/FinetuningRegressor
- SequenceLengthRMSEPlusBERTMLMLossWithReduction
	- masked_token_loss from NeMo no longer does reductions, needs the same fix as pretraining
        - needs vocab parallel mse loss as well

- ExampleModel
- similar issues exist here

### Steps to Reproduce

.

### Error Messages and Logs

```shell

```

### System Information

Environment Details:
- OS: [e.g., Ubuntu 20.04]
- CPU: [e.g., Intel i9-12900K]
- RAM: [e.g., 64GB]

GPU Details:
- GPU Model: [e.g., NVIDIA RTX 4090]
- GPU Memory: [e.g., 24GB]
- CUDA Version: [e.g., 12.1]
- CUDA Driver: [e.g., 525.85.05]
- cuDNN Version: [e.g., 8.9.0]

**Comment by trvachov** [comment]
This sounds like it might need an nvbug. 

- no reduction anywhere (DP), --> new, expected to work in 2.5, 2.6, 2.6.1

All other problems (uses of MSE loss in finetuning) seem to be non-functional on all version of bionemo.

**Comment by trvachov** [comment]
Need to replace torch.mse loss with megatron vocab aware MSE loss.

**Comment by yzhang123** [comment]
https://github.com/NVIDIA/bionemo-framework/pull/959
